lib/Runtime/Language/JavascriptOperators.cpp

```diff
@@ -4818,6 +4818,11 @@ namespace Js
 #endif
         }
 
+        if (result == FALSE && (propertyOperationFlags & PropertyOperation_StrictMode))
```

Comment:
I wonder if this should be the canonical location for throwing on failed-delete-during-strict-mode, rather than having all the individual type handlers do it right now. It would need to also be changed in `OP_DeleteProperty`, but then we could have only 2 places, rather than one per type handler.

Reply:
I'd like to wrap up this change without disrupting other codepaths at the same time. I'll try pulling DeleteProperty higher up the stack but I'll do it separately.

---
In reply to: [172695861](https://github.com/Microsoft/ChakraCore/pull/4786#discussion_r172695861) [](ancestors = 172695861)